keg/image_definition.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2020 SUSE Software Solutions Germany GmbH. All rights reserved.
+#
+# This file is part of keg.
+#
+# keg is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# keg is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with keg. If not, see <http://www.gnu.org/licenses/>
+#
+from datetime import datetime, timezone
+import os
+
+from keg import version, yaml_utils
+from keg.exceptions import KegError
+
+
+class ImageDefinition:
+    """Class for constructing a keg image definition from recipes"""
+
+    def __init__(self, image_name, recipes_root):
+        """Init ImageDefintion with image_name and recipes root path"""
+        self._image_name = image_name
+        self._image_root = os.path.join(recipes_root, 'images')
+        self._data_root = os.path.join(recipes_root, 'data')
+
+    def populate(self):
+        """Parse recipes data and construct wanted image definition"""
+        utc_now = datetime.now(timezone.utc)
+        utc_now_str = utc_now.strftime("%Y-%m-%d %H:%M:%S")
+        self._data = {
+            'generator': 'keg {}'.format(version.__version__),
+            'timestamp': '{}'.format(utc_now_str),
+            'image source path': '{}'.format(self._image_name)
```

Comment:
Singular problem, we need to account for multiple source trees for generating the definition. Also it would be good to have an introductory comment `This kiwi configuration generated by keg......` rather than just `keg`

Reply:
The template produces a comment like that:
```
<!-- Image description generated by {{ data['generator'] }} on {{ data['timestamp'] }} -->
```

Re `image source path`, this is really just the name of the image definition and would read something like `sles/on-demand/15-sp2`. It's also not currently used in the templates.

Reply:
I think more information is valuable thus including `sles/on-demand/15-sp2` in the comment for example would be good.